Added oauth authentication

cptcret opened the pull request:
- Generate API tokens with the connect_to_api script (which references a configuration yaml containing the client id/secret)
- Added token property to the DashboardApi and LookApi objects; store auth token there
- Insert auth token into headers of relevant method http requests

Commit:
Added oauth authentication

added oauth

Removed commented out function

Add config.yml to gitignore

Add config.yml to gitignore

RexGibson commented on lookerpy/apis/dashboard_api.py:
```diff
@@ -28,6 +28,7 @@
 
 from ..configuration import Configuration
 from ..api_client import ApiClient
+from .connect_to_api import connect_to_api as api_connection
```
Why rename?

cptcret replied:
(1) More preference than anything - 'connect_to_api' is a verb, 'api_connection' is a noun.  I'm using the return value of the function, so I thought it'd be a bit clearer to make it a noun.
(2) api_connection is actually a bit misleading, because right now the function just returns the access token.  I'll change this rename to account for that.

cptcret replied:
Removed rename

RexGibson commented on lookerpy/apis/look_api.py:
```diff
@@ -70,25 +71,20 @@ def all_looks(self, **kwargs):
-            params[key] = val
-        del params['kwargs']
+        params = dir(self)
+        print params
```
no print please.

cptcret replied:
Done

RexGibson commented on lookerpy/apis/look_api.py:
```diff
@@ -70,25 +71,20 @@ def all_looks(self, **kwargs):
+        if 'client_secret' in params:
+            query_params['client_secret'] = self.client_secret
 
+        print query_params
```
no print please.

cptcret replied:
Done

RexGibson commented on lookerpy/apis/look_api.py:
```diff
@@ -390,6 +386,7 @@ def run_look(self, look_id, format, **kwargs):
                  returns the request thread.
         """
 
+
```
white space

cptcret replied:
Done

RexGibson commented on lookerpy/apis/look_api.py:
```diff
@@ -411,6 +408,7 @@ def run_look(self, look_id, format, **kwargs):
             raise ValueError("Missing the required parameter `format` when calling `run_look`")
 
         resource_path = '/looks/{look_id}/run/{format}'
+
```
whitespace.

cptcret replied:
Done

RexGibson commented on lookerpy/configuration.py:
```diff
@@ -43,6 +45,16 @@ def _singleton():
 
 
+def _fetch_credentials(configuration_yaml, environment):
+    f = open(configuration_yaml)
```
use a simple context handler here.

RexGibson commented on lookerpy/configuration.py:
```diff
@@ -55,16 +67,28 @@ def __init__(self):
+        self.client_id, self.client_secret = _fetch_credentials(configuration_yaml, environment)
         # Default Base url
-        self.host = "https://looker-stage-vpc.knewton.net:29999/api/3.0"
+        self.host = "https://looker.knewton.net:29999/api/3.0"
```
Can we make this part of the config.yml so that we can move between staging and prod easily?

Commit:
refactored host url into config file (which is in .gitignore), changed yaml reader to context handler

Commit:
Resolved merge conflicts

Commit:
Fixed formatting in files

Fixed weird formatting introduced by auto-formatter

further formatting fixes

Fixed formatting in files

Fixed weird formatting introduced by auto-formatter

further formatting fixes

Fixed formatting in files

Fixed weird formatting introduced by auto-formatter

further formatting fixes

Fixed formatting in files

Fixed weird formatting introduced by auto-formatter

further formatting fixes

RexGibson commented on lookerpy/apis/look_api.py:
```diff
@@ -75,19 +76,20 @@ def all_looks(self, **kwargs):
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
-                    " to method all_looks" % key
+                    " to method create_look_prefetch" % key
```
this looks weird.

cptcret replied:
This block follows the pattern of all the other methods in the class; the 'params = dir(self)' was a change I had made for debugging purposes.
